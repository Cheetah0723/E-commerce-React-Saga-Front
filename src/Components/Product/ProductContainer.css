.ProductContainer{
    flex: 0 0 25%;
}

.card-image{
    cursor: pointer;
}

.button-group-items{
    position: absolute;
    top: 50%;
    color: white;
    font-size: 1rem;
}

.button-group-item:hover,
.button-group-item:active{
    background-color:black;
     color:white;
}

.sales-price{
    color: red;
    text-align: center;
    font-family: 'Montserrat', sans-serif;
    font-size: 0.75rem;
}

@media (max-width: 400px) {
    .ProductContainer{
        width:50%;
        padding:0;
    }

    .item-name{
        text-align: center;
        font-family: 'Roboto', sans-serif;
        font-size: 1.2rem;
    }
    
    .item-color{
        text-align: center;
        font-family: 'Roboto', sans-serif;
        font-size: 1rem;
        color:#4444;
    }
    
    .item-price{
        text-align: center;
        font-family: 'Montserrat', sans-serif;
        font-size: 1.2rem;
        font-weight: bold;
    }
}
@media (min-width: 401px) {
   
    
    
}