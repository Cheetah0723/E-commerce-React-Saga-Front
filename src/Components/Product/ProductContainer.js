import React from 'react';
import history from '../../history';
import { useSelector } from "react-redux"
import "./ProductContainer.css";

export default function ProductContainer({ item }) {
    const containerSize = { width: "100%" }

    const currency = useSelector(state => state.CurrencyReducer).currency

    const price = currency === "AU" ? item.AUprice : item.USprice

    const unit = currency === "AU" ? "AUD" : "USD"

    if (item === undefined) {
        return <div></div>
    }

    return (
        <div className="ProductContainer" key={item.id}>
            <div>
                <img className="card-image" src={require('../../' + item.img)} alt={item.name}
                    onClick={() => {
                        history.push("/product/" + item.id, item.id);
                    }}
                    style={containerSize} />
                <p className="item-name" ><b>{item.name}</b></p>
                <p className="item-color" ><b>{item.color}</b></p>
                {item.salesRate === 1 && <p className="item-price">${price}{unit}
                </p>}
                {item.salesRate < 1 &&
                    <p className="sales-price">SALE ${price * item.salesRate}{unit}
                        ({100 - item.salesRate * 100} % off)</p>}
            </div>
        </div>)
}
