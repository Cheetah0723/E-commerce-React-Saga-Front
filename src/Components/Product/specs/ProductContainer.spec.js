import React from 'react';
import ProductContainer from "../ProductContainer"
import renderer from 'react-test-renderer';
import { useSelector } from "react-redux"

jest.mock('react-redux', () => ({
    useSelector: jest.fn().mockReturnValue({
        addedItems: ["item1"],
        total: 100
    }),
}));

it.skip('renders correctly', () => {
    const item = {
        id: 1,
        name: "Flex Leggings",
        color: "pink",
        img: "product-photos/pink-front.png",
        AUprice: 60,
        USprice: 40,
        category: "bottoms",
        texture: ["56% Nylon", "41% Polyester", "3% Elastane"],
        new: true,
        salesRate: 1,
        img2: "product-photos/pink-back.png",
        size: "XS"
    }
    const test = (arg) => { console.log(arg) }

    const sizeAvailability = [false, false, true, true, true]
    const component = renderer.create(<ProductContainer handleClick={test} addSize={test}
        sizeAvailability={sizeAvailability} item={item}
    />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
