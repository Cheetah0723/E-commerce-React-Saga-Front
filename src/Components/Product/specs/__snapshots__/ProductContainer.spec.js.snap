// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ProductContainer"
  style={
    Object {
      "paddingRight": 5,
    }
  }
>
  <div>
    <img
      alt="Flex Leggings"
      className="card-image"
      onClick={[Function]}
      src="pink-front.png"
      style={
        Object {
          "width": "100%",
        }
      }
    />
    <p
      className="item-name"
    >
      <b>
        Flex Leggings
      </b>
    </p>
    <p
      className="item-color"
    >
      <b>
        pink
      </b>
    </p>
    <p
      className="item-price"
    >
      $
      40
      USD
    </p>
  </div>
</div>
`;
